testing/tests/DevExpress.viz.gauges/barGauge.tests.js

```diff
@@ -574,6 +575,104 @@ QUnit.test('Some values are not changed', function(assert) {
+        assert.strictEqual(group.children[22]._stored_settings.visibility, null, 'line 6 is visible');
+        assert.strictEqual(group.children[23]._stored_settings.visibility, null, 'text 6 is visible');
+        done();
+    }, this);
+});
+
+QUnit.test('label location if the angle is 90 and resolveLabelOverlapping - shift', function(assert) {
+    const done = assert.async();
+    this.$container.dxBarGauge({
+        values: [49, 50, 51],
+
+        resolveLabelOverlapping: 'shift'
+    });
+    const group = this.getBarsGroup();
+
+    group.animationComplete = $.proxy(function() {
+        assert.deepEqual(group.children[2]._stored_settings.points, [194.21667, 31.15986, 21, 7], 'line 1 coords');
+        assert.equal(group.children[3]._stored_settings.x.toFixed(3), 192.275, 'text 1 is coord x');
+        assert.equal(group.children[3]._stored_settings.y.toFixed(3), -1.818, 'text 1 is coord y');
+        assert.deepEqual(group.children[6]._stored_settings.points, [199, 66, 1, 7], 'line 2 coords');
+        assert.equal(group.children[7]._stored_settings.x.toFixed(3), 200.000, 'text 2 is coord x');
+        assert.equal(group.children[7]._stored_settings.y.toFixed(3), -2.000, 'text 2 is coord y');
+        assert.deepEqual(group.children[10]._stored_settings.points, [ 202.48588, 101.08215, -8, 17], 'line 3 coords');
+        assert.equal(group.children[11]._stored_settings.x.toFixed(3), 207.725, 'text 3 is coord x');
+        assert.equal(group.children[11]._stored_settings.y.toFixed(3), -1.818, 'text 3 is coord y');
+        done();
+    }, this);
+});
+
+QUnit.test('label location if the angle is > 90 and resolveLabelOverlapping - shift', function(assert) {
+    const done = assert.async();
+    this.$container.dxBarGauge({
+        values: [80, 79, 78],
+
+        resolveLabelOverlapping: 'shift'
+    });
+    const group = this.getBarsGroup();
+
+    group.animationComplete = $.proxy(function() {
+        assert.deepEqual(group.children[2]._stored_settings.points, [341.22712, 152.47344, 1, 7], 'line 1 coords');
+        assert.equal(group.children[3]._stored_settings.x.toFixed(3), 371.981, 'text 1 is coord x');
+        assert.equal(group.children[3]._stored_settings.y.toFixed(3), 136.345, 'text 1 is coord y');
+        assert.deepEqual(group.children[6]._stored_settings.points, [305.73529, 152.89618, -8, 17], 'line 2 coords');
+        assert.equal(group.children[7]._stored_settings.x.toFixed(3), 370.593, 'text 2 is coord x');
+        assert.equal(group.children[7]._stored_settings.y.toFixed(3), 128.743, 'text 2 is coord y');
+        assert.deepEqual(group.children[10]._stored_settings.points, [ 270.67515, 156.59695, -16, 27], 'line 3 coords');
+        assert.equal(group.children[11]._stored_settings.x.toFixed(3), 368.848, 'text 3 is coord x');
+        assert.equal(group.children[11]._stored_settings.y.toFixed(3), 121.215, 'text 3 is coord y');
+        done();
+    }, this);
+});
+
```

Comment:
@EvgeniiMorozovDevSoft  it's a UI feature so I believe we should add a screenshot tests

Reply:
At the moment, we have discussed that screenshot tests will not help us much if all the alleged cases are covered in the unit test